ci/jenkins/Jenkins_steps.groovy

```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+    return ['Main Jekyll Website': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            utils.init_git()
+            utils.docker_run('ubuntu_cpu_jekyll', 'build_jekyll_docs', false)
+            utils.pack_lib('jekyll-artifacts', 'docs/_build/jekyll-artifacts.tgz', false)
+          }
+        }
+      }
+    }]
+}
+
+
+// This is for publishing the full website
+def docs_prepare() {
+    return ['Prepare for publication of the full website': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            utils.init_git()
+
+            unstash 'jekyll-artifacts'
+            unstash 'c-artifacts'
+            unstash 'python-artifacts'
+            unstash 'r-artifacts'
+            unstash 'julia-artifacts'
+            unstash 'scala-artifacts'
+            unstash 'java-artifacts'
+            unstash 'clojure-artifacts'
+
+            utils.docker_run('ubuntu_cpu_jekyll', 'build_docs', false)
+
+            // only stash if we're going to unstash later
+            // utils.pack_lib('full_website', 'docs/_build/full_website.tgz', false)
+
+            // archive so the publish pipeline can access the artifact
+            archiveArtifacts 'docs/_build/full_website.tgz'
+          }
+        }
+      }
+    }]
+}
+
+
+// This is for the website and the Python API only
+def docs_prepare_python_only() {
+    return ['Prepare for publication of the full website': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            utils.init_git()
+
+            unstash 'jekyll-artifacts'
+            unstash 'python-artifacts'
+
+            // Prepare website and Python API docs
+            utils.docker_run('ubuntu_cpu_jekyll', 'build_docs_small', false)
+
+            // Publish preview to S3
+            sh "cd docs/_build/html && ci/other/ci_deploy_doc.sh ${env.BRANCH_NAME} ${env.BUILD_NUMBER}"
+          }
+        }
+      }
+    }]
+}
+
+
+def docs_archive() {
+    return ['Archive the full website': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            archiveArtifacts 'docs/_build/full_website.tgz'
+          }
+        }
+      }
+    }]
+}
+
+
+// This is for the full website
+def docs_publish() {
+    return ['Publish the full website': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            //utils.init_git()
+            //unstash 'full_website'
+            //sh 'tar -xzf docs/_build/full_website.tgz --directory .'
+            // TODO: Make sure this 'test-website-publish' understand the new structure
+            try {
+              build 'test-website-publish-master'
+            }
+            catch (Exception e) {
+               println(e.getMessage())
```

Comment:
println("Building docs failed", e.getMessage()) or something to that effect (syntax there is incorrect).